### PR TITLE
Add simple bar chart using mock data

### PR DESCRIPTION
**Pull request by bennettbuchanan**
1. Add a simple bar chart using mock data.
2. https://trello.com/c/B74eKXHy/29-proof-of-concept-of-react-d3-app
    https://trello.com/c/lo8Vrbrj/25-render-simple-chart
3. This introduces the use of D3, so perhaps we want to update the onboarding docs with some tutorials or explanation. Considering this is a proof-of-concept, I have left those alone for now.
Here's a screenshot of the chart that is rendered from the mock data:
screen shot 2018-02-25 at 8 33 22 pm
4. @erikeldridge

### CODE REVIEW

File: django/congressionaldata/webapp/components/App.jsx [hunk 1]
```diff
@@ -3,6 +3,8 @@ import request from 'superagent';
 import { Alert } from 'react-bootstrap/lib';
 
 import safeJSONParse from '../util/safeJSONParse';
+import BarChart from './charts/BarChart.jsx';
```
Comment (erikeldridge) [inline django/congressionaldata/webapp/components/App.jsx]: Nice encapsulation, and thx for the plural charts dir :)

File: django/congressionaldata/webapp/components/App.jsx [hunk 2]
```diff
@@ -60,6 +62,12 @@ class App extends Component {
         return (
             <div>
                 <h1>Congressional Data</h1>
+                <div className='chart'>
```
Comment (erikeldridge) [inline django/congressionaldata/webapp/components/App.jsx]: minor: seems redundant to wrap an element called "BarChart" in a class called "chart". Can BarChart define a class? I dropped a related comment on the css.

File: django/congressionaldata/webapp/components/App.jsx [hunk 3]
```diff
@@ -60,6 +62,12 @@ class App extends Component {
...
+                    <BarChart data={mockData}
```
Comment (erikeldridge) [inline django/congressionaldata/webapp/components/App.jsx]: nice api!

File: django/congressionaldata/webapp/components/charts/BarChart.jsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+    componentDidMount() {
+        this.createChart()
+    }
+
+    createChart = () => {
+        const { width, height, data } = this.props;
+        const margin = {top: 20, right: 0, bottom: 110, left: 40};
+
+        const x = scaleBand()
```
Comment (erikeldridge) [inline django/congressionaldata/webapp/components/charts/BarChart.jsx]: This is great. D3 and React semantics work well together. Nice choice of tooling and proj structure.

File: django/congressionaldata/webapp/css/app.css [hunk 5]
```diff
@@ -5,3 +5,7 @@ h1 {
 .contribution-list {
     margin: 20px;
 }
+
+.chart rect {
+  fill: steelblue;
```
Comment (erikeldridge) [inline django/congressionaldata/webapp/css/app.css]: minor: seems we're breaking the chart abstraction by styling an implementation detail. Maybe give the rect elements a class BarChart consumers can hook into, ie treat styling like part of the BarChart api?

Also, excellent use of a standard color ;)

File: django/congressionaldata/webapp/components/App.jsx [hunk 6]
```diff
@@ -3,6 +3,8 @@ import request from 'superagent';
 import { Alert } from 'react-bootstrap/lib';
 
 import safeJSONParse from '../util/safeJSONParse';
+import BarChart from './charts/BarChart.jsx';
+import mockData from '../mock/data.json';
```
Comment (erikeldridge) [inline django/congressionaldata/webapp/components/App.jsx]: Nice separation of presentation and data. I can imagine a discrete task to wire this up w live data.

File: django/congressionaldata/webapp/components/App.jsx (hunk 2 above)
Comment (bennettbuchanan) [reply django/congressionaldata/webapp/components/App.jsx]: Updated.

Comment (bennettbuchanan) [reply django/congressionaldata/webapp/components/App.jsx]: Updated.

File: django/congressionaldata/webapp/css/app.css (hunk 5 above)
Comment (bennettbuchanan) [reply django/congressionaldata/webapp/css/app.css]: I like that idea. I updated the BarChart component to take a property `barColor` instead and removed this CSS class.

File: django/congressionaldata/webapp/components/App.jsx (hunk 1 above)
Comment (bennettbuchanan) [reply django/congressionaldata/webapp/components/App.jsx]: The `charts` dir is an aspiration—more charts! 😸

**Comment by bennettbuchanan** [comment]
@erikeldridge I pushed another commit (5c877bf) addressing your comments. Hopefully this helps with with another round of review. I will squash that commit if it looks good to you, then we can merge as a single commit. Thanks!

**Pull request merged**